zcash_client_backend/src/data_api/wallet.rs

```diff
@@ -499,14 +529,14 @@ where
 /// to fall back to the transparent receiver until full Orchard support is implemented.
 #[allow(clippy::too_many_arguments)]
 #[allow(clippy::type_complexity)]
-pub fn create_proposed_transaction<DbT, ParamsT, InputsErrT, FeeRuleT>(
+pub fn create_proposed_transaction<DbT, ParamsT, InputsErrT, FeeRuleT, N>(
```

Comment:
> so this floating type is just to avoid adding unnecessary constraints on DbT

Absolutely not; this universally quantified type is to enforce the correctness of the implementation. Because `N` is universally quantified, the implementation of this function cannot use any information about `N`. This approach is better than hardcoding the result type to `Infallible` because it allows the caller to use it both in contexts where `N` is known to be `Infallible` and in contexts where the caller is able to handle errors.

Reply:
What's more, because `N` is universally quanitfied, the caller has a proof that this function cannot return errors of type `N`, and can adapt its implementation to the result accordingly.

Reply:
> Because N is universally quantified, the implementation of this function cannot use any information about `N`.

This is a constraint on the _function implementor_ (us), not on the downstream user, so describing it as a constraint (the "forces implementations to ignore" language) in the changelog (which is written for downstream users, not us) is what I was raising as an issue.

Reply:
> This approach is better than hardcoding the result type to `Infallible`

I agree; having the generic parameter is correct in this situation (where the function does not need to use that parameter). But that is separate from whether this is a constraint on downstream users (it is not).

Reply:
Except that "forces _implementations_ to ignore" is exactly correct. It doesn't force anything on the caller, and the caller can deduce from the parameterization that this constraint is correctly observed by _every_ implementation.

Reply:
I guess we'll have to agree to disagree on this. I find this language confusing and misleading to downstream users, because "implementations" is not going to be interpreted the way you are using it here.

Reply:
Would you prefer "implementations of this trait"?

Reply:
This is the core problem: this isn't a trait method! It's a concrete function with only one implementation, by us.